### PR TITLE
mes-1774: Rehydration of input field for new email

### PR DESCRIPTION
**Pull request by MarkPatt89**
## Description and relevant Jira numbers

- Rehydrate new email field.
- Initialise a default selection of provided email, if one is available.
- Initialise a default selection of new email, if a provided email is unavailable.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/modules/tests/communication-preferences/communication-preferences.selector.ts [hunk 1]
```diff
@@ -5,4 +5,5 @@ export const getCommunicationPreferenceUpdatedEmail
     communicationPreferences.updatedEmail ? communicationPreferences.updatedEmail : '';
 
 export const getCommunicationPreferenceType
-  = (communicationPreferences: CommunicationPreferences) => communicationPreferences.communicationMethod;
+  = (communicationPreferences: CommunicationPreferences) =>
+  communicationPreferences.communicationMethod ? communicationPreferences.communicationMethod : '';
```
Comment (szabolcs-keresztes) [inline src/modules/tests/communication-preferences/communication-preferences.selector.ts]: Would be a good idea to use `lodash.get` with the default value of an empty string

File: src/pages/communication-page/communication.ts [hunk 2]
```diff
@@ -52,21 +52,24 @@ interface CommunicationPageState {
 })
 export class CommunicationPage extends BasePageComponent {
 
-  readonly providedEmail = 'providedEmail';
-  readonly newEmail = 'newEmail';
+  readonly providedEmail = 'Provided';
```
Comment (szabolcs-keresztes) [inline src/pages/communication-page/communication.ts]: If you declare these variables as static, they become constants and won't be changeable even from the constructor.

File: src/pages/communication-page/communication.ts [hunk 3]
```diff
@@ -153,55 +156,108 @@ export class CommunicationPage extends BasePageComponent {
...
-    this.communicationChoice = communicationChoice;
+  setCommunicationType(communicationChoice: string, emailType: string = null) {
+    this.communicationType = communicationChoice;
+    this.emailType = emailType;
   }
 
-  isSelected(communicationChoice: string) {
-    return (this.communicationChoice === communicationChoice);
+  isProvidedEmailSelected() {
+    return (this.communicationType === 'Email') && (this.emailType === this.providedEmail);
```
Comment (szabolcs-keresztes) [inline src/pages/communication-page/communication.ts]: No need for the brackets.

Same applies to the other 3 methods below.

File: src/pages/communication-page/components/provided-email/provided-email.ts [hunk 4]
```diff
@@ -1,12 +1,18 @@
...
+import { FormControl, FormGroup, Validators } from '@angular/forms';
 
 @Component({
   selector: 'provided-email',
   templateUrl: 'provided-email.html',
 })
 export class ProvidedEmailComponent {
 
-  readonly providedEmail = 'providedEmail';
+  readonly providedEmail = 'Email';
```
Comment (szabolcs-keresztes) [inline src/pages/communication-page/components/provided-email/provided-email.ts]: static - see comment above

File: src/pages/communication-page/communication.html [hunk 5]
```diff
@@ -19,38 +19,43 @@
...
+              </ion-row>
+              <ion-row class="communication-option">
+                <div class="validation-bar communication-validation-bar" [class.ng-invalid]="invalid"></div>
+                <input type="radio" id="support-centre" class="gds-radio-button"
+                  formControlName="radioCtrl" name="radioCtrl" (click)="setCommunicationType('Support Centre')"
+                  [value]="isSupportCentreSelected()" [checked]="isSupportCentreSelected()">
+                <label for="support-centre" class="radio-label">By calling the support centre</label>
+              </ion-row>
+              <ion-row class="validation-message-row" align-items-center>
+                <div class="validation-text"></div>
```
Comment (rossfauldsbjss) [inline src/pages/communication-page/communication.html]: is there no validation text to be shown, if for example they load form and hit submit without picking.
( or is the text on another element)?

File: src/pages/communication-page/communication.ts [hunk 6]
```diff
@@ -153,55 +156,131 @@ export class CommunicationPage extends BasePageComponent {
...
+    Object.keys(this.form.controls).forEach(controlName => this.form.controls[controlName].markAsDirty());
+    if (this.form.valid) {
+      this.navController.push('WaitingRoomPage');
+    }
   }
 
-  dispatchCandidateChoseProvidedEmail(emailType: string) {
-    this.setCommunicationType(emailType);
+  dispatchCandidateChoseProvidedEmail(communicationType: string) {
+    this.setCommunicationType(communicationType, this.providedEmail);
```
Comment (rossfauldsbjss) [inline src/pages/communication-page/communication.ts]: dispatchCandidateChoseProvidedEmail is passed a string - but it is only ever 'Email', so it feels like the parameter is unnecessary.

**Pull request merged**